File: misc/sdl_test.lua
#!/usr/bin/env luajit


local ffi = require("ffi")
local sdl = require("sdl2")


function test(w,h,title)
	
	
	local rect = ffi.typeof("SDL_Rect")({ x = 0, y = 0, w = 100, h = 100 })
	local event = ffi.typeof("SDL_Event")()
	
	local run = true
	
	
	
	
	sdl.init(sdl.INIT_VIDEO)
	
	window = sdl.createWindow(title, 0,0, w,h, 0)
	
	screen = sdl.getWindowSurface(window)
	
	while run do
	
		rect.x = math.random(0, w-100)
		rect.y = math.random(0, h-100)
	
		sdl.fillRect(screen, rect, math.random(0, 0xFFFFFF))
		
		sdl.updateWindowSurface(window)
		
		while sdl.pollEvent(event) == 1 do
			if event.type == sdl.QUIT then
				run = false
			end
		end
		
	end
	
end


test(800,600, "hello world")

File: misc/sdl_test.c


#include <stdio.h>
#include <stdlib.h>
#include <SDL2/SDL.h>


void main() {
	SDL_Window *window;
	SDL_Surface *screen;
	SDL_Rect rect;
	SDL_Event event;

	uint run = 1;

	rect.w = 100;
	rect.h = 100;

	SDL_Init(SDL_INIT_VIDEO);

	window = SDL_CreateWindow("hello world!",0, 0, 800, 600, 0);

	screen = SDL_GetWindowSurface(window);

	while (run) {

		rect.x = rand() % 700;
		rect.y = rand() % 500;

		SDL_FillRect(screen, &rect, rand());

		SDL_UpdateWindowSurface(window);

		while (SDL_PollEvent(&event)) {
			if (event.type == SDL_QUIT){
				run = 0;
			}
		}

	}

}

